src/javis_viewer.jl

```diff
@@ -286,3 +279,124 @@ function _pluto_viewer(video::Video, frames::Int, objects::Vector)
     arr = collect(get_javis_frame(video, objects, frame) for frame in 1:frames)
     return arr
 end
+
+"""
+    setup_stream(livestreamto=:local; protocol="udp", address="0.0.0.0", port=14015, twitch_key="")
+
+Sets up the livestream configuration.
+**NOTE:** Twitch not fully implemented, do not use.
+"""
+function setup_stream(
+    livestreamto::Symbol = :local;
+    protocol::String = "udp",
+    address::String = "0.0.0.0",
+    port::Int = 14015,
+    twitch_key::String = "",
+)
+    StreamConfig(livestreamto, protocol, address, port, twitch_key)
+end
+
+"""
+    cancel_stream()
+
+Sends a `SIGKILL` signal to the livestreaming process. Though used internally, it can be used stop streaming.
+However this method is not guaranted to end the stream on the client side.
+"""
+function cancel_stream()
+    #todo explore better ways of searching and killing processes
+
+    # kill the ffmpeg process
+    # ps aux | grep ffmpeg | grep stream_loop | awk '{print $2}' | xargs kill -9
+    try
+        println("Checking for existing stream....")
+        run(
+            pipeline(
+                `ps aux`,
+                pipeline(`grep ffmpeg`, pipeline(`grep stream_loop`, `awk '{print $2}'`)),
+            ),
+        )
+    catch
+        return @warn "Not Streaming Anything Currently"
+    end
+
+    run(
+        pipeline(
+            `ps aux`,
+            pipeline(
+                `grep ffmpeg`,
+                pipeline(`grep stream_loop`, pipeline(`awk '{print $2}'`, `xargs kill -9`)),
+            ),
+        ),
+    )
+    return "Livestream Cancelled!"
+end
+
+"""
+    _livestream(streamconfig, framerate, width, height, pathname)
+
+Internal method for livestreaming 
+"""
+function _livestream(
+    streamconfig::StreamConfig,
+    framerate::Int,
+    width::Int,
+    height::Int,
+    pathname::String,
+)
+    cancel_stream()
+
+    livestreamto = streamconfig.livestreamto
+    twitch_key = streamconfig.twitch_key
+
+    if livestreamto == :twitch && isempty(twitch_key)
+        return error("Please enter your twitch stream key")
+    end
+
+    command = [
+        "-stream_loop", # loop the stream -1 times i.e. indefinitely
+        "-1",
+        "-r", # frames per second
+        "$framerate",
+        "-an",  # Tells FFMPEG not to expect any audio
+        "-loglevel", # show only ffmpeg errors
+        "error",
```

Comment:
should this depend on the `ffmpeg_loglevel` of the `render` function?

Reply:
I had actually thought of that. But since streaming is synchronous, having it is set to `error` by default is beneficial incase the stream fails

Reply:
Oh yeah didn't realize the other one is `panic`. We might wanna have some `loglevel` struct later with different possibilities at some point. But it's okay for now.